torch/csrc/jit/tensorexpr/ir_mutator.cpp

```diff
@@ -515,11 +518,11 @@ Stmt* StmtClone::mutate(const AtomicAdd* v) {
 }
 
 Stmt* StmtClone::mutate(const Allocate* v) {
-  return new Allocate(v->buffer_var(), v->dtype(), v->dims());
+  return new Allocate(new Buf(v->buffer_var(), v->dims(), v->dtype()));
```

Comment:
If we had `Allocate` and `Free` statements accessing the same `Buf` and clone both of them, the clones would refer to different `Buf`s, because when we clone `Allocate` we create a new `Buf` object. We should be using `v->buf()` instead here - `Buf` are expressions and expressions are immutable, so we don't need to create copies of them (in contrast to statements, which are mutable, and for which we do need to create copies).